### PR TITLE
chore(api): fix GraphQLModelBasedTests to async

### PR DESCRIPTION
**Pull request by manaswi223**
*Issue #, if available:*

*Description of changes:*
Fixed `GraphQLModelBasedTests.swift` from old apis to new async apis except for a test `testConcurrentSubscriptions()` 
Screen Shot 2022-08-24 at 12 06 06 AM

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [x] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ enum PostStatus {
     PUBLISHED
 }
 
-type Post @model {
+type Post @model @auth(rules: [{ allow: public }]) {
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/README.md:37]: is this needed? I thought in V1 transformer, it is not deny by default and models can be accessed via APIKey if there are no `@auth` rules on it

**Comment by lawmicha** [comment]
Looks like there are a few hiding in extensions of GraphQLModelBasedTests that need fixing 
- testConcurrentSubscriptions
- testPaginatedListFetch
- testPaginatedListFetchValidationError
- testFetchListOfCommentsFromPostFails

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLModelBasedTests.swift [hunk 2]
```diff
@@ -453,116 +343,94 @@ class GraphQLModelBasedTests: XCTestCase {
...
-        let connectedInvoked = expectation(description: "Connection established")
-        let disconnectedInvoked = expectation(description: "Connection disconnected")
-        let completedInvoked = expectation(description: "Completed invoked")
-        let progressInvoked = expectation(description: "progress invoked")
-
-        let operation = Amplify.API.subscribe(
-            request: .subscription(of: Post.self, type: .onDelete),
-            valueListener: { event in
-                switch event {
+    func testOnDeletePostSubscriptionWithModel() async throws {
```
Comment (lawmicha) [inline AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLModelBasedTests.swift:351]: add new line between the func and the previous func

File: AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/README.md (hunk 1 above)
Comment (manaswi223) [reply AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/README.md]: actually, I was getting warnings to use @auth rules, so when I updated the warnings stopped popping, no worrries - I've discarded the @auth changes.

**Comment by lawmicha** [comment]
this can be closed as dup to #2197

**Pull request closed without merging**